### PR TITLE
fix: embedded credit card fields are styled consistently with Sequoia form template

### LINKED ISSUE #4729: fix: Embedded credit card fields should be styled consistently in Sequoia form template
# Bug Report

## User Story
As an admin, I want embedded credit card fields to be styled consistently with other Sequoia template fields so that users can have a cohesive donation experience, regardless of what payment gateways are enabled.

## Details
Some credit card gateways use fields embedded as iframes to collect card information. These fields should be styled consistently with the rest of the Sequoia template inputs.

## Current Behavior
Embedded gateway fields are not styled consistently within the Sequoia form template.

## Expected Behavior
These fields should be styled to match Figma designs, and other inputs in the Sequoia form template. 

See Figma: https://www.figma.com/file/jaVXDA4wjQD9OqTi0l9KQe/Form-Templates?node-id=1882%3A44&viewport=-259%2C1586%2C0.5

## Bug Type
- [ ] This bug describes functionality that once worked as expected in version X.X.X.
- [x] This bug describes functionality that never worked as expected.
- [ ] I am not sure whether this functionality ever worked as expected.

## Acceptance Criteria
- [ ] Credit card field appearance matches Figma design
- [ ] Credit card fields are styled consistently, regardless of gateway
- [ ] Fields work as expected and payment processing is unaffected

### PR DESCRIPTION
**Pull request by henryholtgeerts**
## Description
Resolves #4729 
This PR improves styling of embedded credit card fields in the Sequoia form template. Perviously, when credit card fields were securely embedded as iframes, they did not inherit styling applied to other input fields in the Sequoia form template. Now, these fields are styled to match other text inputs.

The PR uses css and existing filters to define styles for these fields. While parts of the input could be styled through the same stylesheet that styles other inputs in the Sequoia template, the input fonts had to be styled by hooking into existing filters for stripe styles, which are used in creating the fields via the Stripe API on the frontend.

## Affects
This PR impacts Sequoia assets and the Sequoia Actions class.

## What to test
Enable PayPal Pro and Stripe credit card gateways. As a user select each gateway. Are the input fields styled correctly? Do they match other input fields in the Sequoia form template?

## Screenshots:
Screen Shot 2020-05-15 at 12 18 51 PM

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by DevinWalker** [comment]
Hey @henryholtgeerts thanks for the first pass at this. Currently, the CC fields look better but I'd like them to look a little nicer. Here's a concept I put together that's in Figma. Can you please take a second pass over it?

**Comment by henryholtgeerts** [comment]
Style changes have been made to match Figma design put together by @DevinWalker, see screenshot: 

Screen Shot 2020-05-15 at 3 14 44 PM

Still need to test with addional Payment Gateways (Square and Authorize.net)

**Comment by henryholtgeerts** [comment]
This PR has been tested with Square and Authorize.net gateways, and works as expected.

**Comment by ravinderk** [comment]
@henryholtgeerts I tested with Square and noticed a style conflict. You can see inside the card field there are gray and white areas. 

image

**Comment by DevinWalker** [comment]
@henryholtgeerts here is what the Stripe Modal gateway should display when it's toggled:

**Comment by DevinWalker** [comment]
@henryholtgeerts this is how Stripe Credit Card on site looks if you're using the "Single Field" option:





This needs to be resolved as well as all other gateways using single fields:



It is incredibly important we ensure all Stripe gateways look good and function properly. 

The JS issue I mentioned in the video is particularly concerning. @mehul0810 may be able to provide insight into that.

**Comment by henryholtgeerts** [comment]
The following issues have been addressed:
- [x] Styling single line fields
- [x] Styling BECS and SEPA gateways
- [x] Adding Stripe checkout offsite redirection content
- [x] Stripe Elements viewport warning in browser console

Screen Shot 2020-05-21 at 4 11 31 PM

Screen Shot 2020-05-21 at 4 11 41 PM

Screen Shot 2020-05-21 at 4 11 57 PM

Screen Shot 2020-05-21 at 4 21 16 PM

Note that the following bugs still need to be addressed:
- [ ] Stripe fields not consistently loading (multiple divs with same ID)
- [ ] Stripe fields not loading Montserrat font consistently
- [ ] JS error pointed out in @DevinWalker's video walkthrough

**Comment by henryholtgeerts** [comment]
Requested changes have been made to this PR, and it is ready for review. Latest screenshots of various different Stripe gateways:

Screen Shot 2020-05-22 at 1 29 56 PM

Screen Shot 2020-05-22 at 1 30 09 PM

**Comment by DevinWalker** [comment]
@henryholtgeerts we're making some good progress here. Please review this video for my findings: https://www.loom.com/share/2535071d706445ed9c1c95d444fc58da

**Comment by DevinWalker** [comment]
@henryholtgeerts here are the placeholder content concepts for Plaid and iDEAL:





Note: The following gateways should use the bank account icon in the toggle:

1. SEPA
2. BECA
3. iDEAL
4. Plaid

**Comment by henryholtgeerts** [comment]
Issues to resolve in this PR:
- [ ] Decrease padding beneath CC fields
- [ ] Ensure that select fields match text input field height
- [ ] Resolve Square styling issues

Issues to tack in separate issue/PR:
- [ ] Google Pay compatibility (hide donate now button)
- [ ] Move transaction fee coverage notice to bottom (above Donate Now button)
- [ ] Plaid compatibility (handle modal donation, show notice copy/image)
- [ ] iDeal compatibility (show offsite redirection notice)

**Comment by henryholtgeerts** [comment]
@DevinWalker Running into a wall getting Square styling to match exactly, since there's no existing GiveWP api for that. As such, it uses system font, and fields are slightly larger. Here's a screenshot: 

Screen Shot 2020-05-26 at 2 54 28 PM

Outside of this issue, this PR is ready for review.

**Comment by DevinWalker** [comment]
@henryholtgeerts I'm still seeing issues...

### Square Credit Card



1. The bottom padding is still not even with the top padding and is too much. Please play closer attention to this as it's been at least 3 or 4 times I have had to point this out.
2. If the Square CC fields need to be a larger font size then make the other fields the same size if possible.
3. The "State" dropdown select is not the same size as the "Postal Code" field
4. The "Cardholder Name" field is not as large as the expiration date field. Field heights should match. Again, this is something I have mentioned multiple times. 

### Stripe Credit Card



1. The bottom padding is still not correct
2. The "Cardholder Name" field border color does not match the Stripe CC fields. This is the third or fourth time I've mentioned this...

### SEPA



1. The bottom padding is still not correct
2. The borders are not the same color between the bank number field and the address fieldset
3. When the "Mandate Acceptance Text" is set to display the margins and padding should be even. Currently there's inconsistencies when you toggle this option on and off.

### BECS 



1. The bottom padding is still not correct
2. The borders are not the same color between the bank number field and the address fieldset
3. When the "Mandate Acceptance Text" is set to display the margins and padding should be even. Currently there's inconsistencies when you toggle this option on and off.

**Comment by henryholtgeerts** [comment]
@DevinWalker The final notes you outlined have been incorporated, here are screenshots for reference:

Re-styled Square fields, to ensure consistent styling with Square embedded inputs
Screen Shot 2020-05-27 at 5 28 21 PM

Embedded Stripe CC fields border color set to match Sequoia input border color (change carried over to all embedded Stripe fields):
Screen Shot 2020-05-27 at 5 33 02 PM

Spacing made consistent above/below mandatory acceptance text (regardless of it being present or not): 
Screen Shot 2020-05-27 at 5 44 23 PM

Screen Shot 2020-05-27 at 5 44 34 PM

Above screenshots also reflect improvements to CC wrapper padding, made to match top/bottom and left/right.

**Comment by DevinWalker** [comment]
Looks great now. Thanks!

File: includes/gateways/stripe/includes/payment-methods/class-give-stripe-checkout.php [hunk 1]
```diff
@@ -63,6 +65,38 @@ public function __construct() {
 
 		}
 
+		public function output_redirect_notice() {
```
Comment (ravinderk) [inline includes/gateways/stripe/includes/payment-methods/class-give-stripe-checkout.php:68]: @henryholtgeerts This function does not have a description.

Comment (ravinderk) [reply includes/gateways/stripe/includes/payment-methods/class-give-stripe-checkout.php:68]: fixed: https://github.com/impress-org/givewp/commit/aba541debb24b5eb9855d9daaa292b8e1ec6096c

Comment (ravinderk) [reply includes/gateways/stripe/includes/payment-methods/class-give-stripe-checkout.php:68]: @henryholtgeerts what is use of this function? This function functionality seems Sequoia form template specific only

File: src/Views/Form/Templates/Sequoia/assets/css/tributes.scss [hunk 2]
```diff
@@ -1,83 +0,0 @@
-.give-tributes-dedicate-donation {
```
Comment (ravinderk) [inline src/Views/Form/Templates/Sequoia/assets/css/tributes.scss:1]: @henryholtgeerts Did not intentionally delete this file?

File: assets/src/js/frontend/give-stripe-sepa.js [hunk 3]
```diff
@@ -19,11 +19,10 @@ document.addEventListener( 'DOMContentLoaded', function( e ) {
 	const fontStyles = [];
 	const preferredLocale = give_stripe_vars.preferred_locale;
 	const formWraps = document.querySelectorAll( '.give-form-wrap' );
-	const fontIterator = Object.entries( give_stripe_vars.element_font_styles );
```
Comment (ravinderk) [inline assets/src/js/frontend/give-stripe-sepa.js:22]: @mehul0810 it is a necessary change: https://github.com/impress-org/givewp/commit/e059ab1fe5175e30a3cc262decef3414e756a37f

File: assets/src/js/frontend/give-stripe.js [hunk 4]
```diff
@@ -20,11 +22,10 @@ document.addEventListener( 'DOMContentLoaded', function( e ) {
 	const fontStyles = [];
 	const preferredLocale = give_stripe_vars.preferred_locale;
 	const formWraps = document.querySelectorAll( '.give-form-wrap' );
-	const fontIterator = Object.entries( give_stripe_vars.element_font_styles );
```
Comment (ravinderk) [inline assets/src/js/frontend/give-stripe.js:23]: @mehul0810 it is a necessary change: https://github.com/impress-org/givewp/commit/e059ab1fe5175e30a3cc262decef3414e756a37f

**Pull request merged**